File: app/src/main/java/com/nfsindustries/vsd/VSDJNI.java
package com.nfsindustries.vsd;

public class VSDJNI {

    static{
        try{
            System.loadLibrary("VSDJNI");
        }catch(UnsatisfiedLinkError e){
            e.printStackTrace();
        }
    }

    // --- Native VSD method
    public native double processAudio(double[] inputArray);
}

File: app/src/main/java/com/nfsindustries/vsd/VSDJNI.kt
package com.nfsindustries.vsd

import android.util.Log

class VSDJNI {

    // --- Native VSD method
    external fun processAudio(inputArray: DoubleArray): Double

    companion object {

        init {
            try {
                System.loadLibrary("VSDJNI")
            } catch (e: UnsatisfiedLinkError) {
                Log.e(LOG_TAG, e.message)
            }

        }
    }
}
